solutions/cf-20C.mdx

```diff
@@ -76,4 +76,57 @@ int main() {
 
 </CPPSection>
 
+<PySection>
+	
+```py
+
+import heapq
+import sys
+
+
+def dijkstra(graph, source):
+    vis = [False] * (len(graph) + 1)  # initializing each node as unvisited
+    dist = [sys.maxsize] * (len(graph) + 1)  # initializing distance of source to all vertices as +infinity
+    par = [-1] * (len(graph) + 1)  # initializing parent of each node as -1
+    mnpq = []  # minimum priority queue
+    dist[source] = 0  # setting distance of source to itself as 0
+    heapq.heappush(mnpq, (0, source))  # pushing (distance from source, node) into mnpq
+    while mnpq:
+        w1, cur = heapq.heappop(mnpq)  # popping minimum distance from source,node
+        if vis[cur]:
+            continue
+        for v, w in graph[cur]:  # if not visited, iterate through all neighbours of current node
+            if vis[v]:
+                continue
+            if (w + dist[cur]) < dist[v]:  # if (weight between cur and v + dist of cur from source) < dist of v from source
+                dist[v] = w + dist[cur]  # update dist of v
+                par[v] = cur  # update par of v as cur
+            heapq.heappush(mnpq, (dist[v], v))  # push updated dist, node v into mnpq
+        vis[cur] = True  # mark cur visited
```

Comment:
basically all of these as well
i'd just put a comment above the function like "run's dijkstra's on a graph, returning [what does it return?]"

Reply:
Removed them